src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -162,4 +161,34 @@ private function registerReceivers(ContainerBuilder $container)
 
         $container->getDefinition('messenger.receiver_locator')->replaceArgument(0, $receiverMapping);
     }
+
+    private function registerMiddlewares(ContainerBuilder $container)
+    {
+        if (!$container->hasParameter('messenger.buses_names')) {
+            return;
+        }
+
+        $busesNames = $container->getParameter('messenger.buses_names');
+        $middlewaresPerBuses = array();
+
+        foreach ($container->findTaggedServiceIds('messenger.bus_middleware') as $serviceId => $tags) {
+            foreach ($tags as $tag) {
+                $priority = $tag['priority'] ?? 0;
+                $buses = isset($tag['bus']) ? explode(',', $tag['bus']) : $busesNames;
```

Comment:
Not fond of the special syntax, AFAIK we are used to require adding as much tags as needed e.g.
```yaml
App\SomeMiddleware:
    tags:
        - { name: "messenger.bus_middleware", bus: "query" }
        - { name: "messenger.bus_middleware", bus: "command" }